Common/Securities/CurrencyConversion/SecurityCurrencyConversion.cs

```diff
@@ -0,0 +1,283 @@
+                    continue;
+                }
+
+                return new SecurityCurrencyConversion(sourceCurrency, destinationCurrency, new List<Step>(1)
+                {
+                    CreateStep(potentialConversionRateSymbol, inverted, securitiesBySymbol, makeNewSecurity)
+                });
+            }
+
+            // Search for 2 leg conversions
+            foreach (var potentialConversionRateSymbol1 in allSymbols)
+            {
+                if (!potentialConversionRateSymbol1.PairContainsCurrency(sourceCurrency))
+                {
+                    continue;
+                }
+
+                var middleCurrency = potentialConversionRateSymbol1.CurrencyPairDual(sourceCurrency);
+
+                foreach (var potentialConversionRateSymbol2 in allSymbols)
+                {
+                    bool secondStepInverted;
+                    if (!SymbolConvertsCurrencies(potentialConversionRateSymbol2,
+                        middleCurrency,
+                        destinationCurrency,
+                        out secondStepInverted))
+                    {
+                        continue;
+                    }
+
+                    var steps = new List<Step>(2);
+
+                    // Step 1
+                    string baseCurrency;
+                    string quoteCurrency;
+
+                    CurrencyPairUtil.DecomposeCurrencyPair(
+                        potentialConversionRateSymbol1,
+                        out baseCurrency,
+                        out quoteCurrency);
+
+                    steps.Add(CreateStep(potentialConversionRateSymbol1,
+                        sourceCurrency == quoteCurrency,
+                        securitiesBySymbol,
+                        makeNewSecurity));
+
+                    // Step 2
+                    steps.Add(CreateStep(potentialConversionRateSymbol2,
+                        secondStepInverted,
+                        securitiesBySymbol,
+                        makeNewSecurity));
+
+                    return new SecurityCurrencyConversion(sourceCurrency, destinationCurrency, steps);
+                }
+            }
+
+            throw new ArgumentException(
+                $"No conversion path found between source currency {sourceCurrency} and destination currency {destinationCurrency}");
+        }
+
+        /// <summary>
+        /// Checks whether a symbol converts between two currencies, and whether it does so in inverse or not
+        /// </summary>
+        /// <param name="symbol">The symbol to check for</param>
+        /// <param name="sourceCurrency">The currency to convert from</param>
+        /// <param name="destinationCurrency">The currency to convert to</param>
+        /// <param name="inverted">The parameter that contains whether the conversion is inverted, or False if the conversion is not possible with the given symbol</param>
+        /// <returns></returns>
+        private static bool SymbolConvertsCurrencies(
```

Comment:
This method is decomposing `Symbol symbol` 3 times on each call it does, we should do it just once